processor/src/main/java/org/mapstruct/ap/internal/model/source/selector/TypeSelector.java

```diff
@@ -197,11 +201,13 @@ else if ( matchingParameterAssignmentVariants.size() == 1 ) {
             // we had some matching assignments before, but when checking the parameter names we can't find an
             // appropriate one, in this case the user must chose identical parameter names for the mapping and lifecycle
             // method
-            messager.printMessage(
-                selectedMethod.getExecutable(),
-                Message.LIFECYCLEMETHOD_AMBIGUOUS_PARAMETERS,
-                mappingMethod
-            );
+            if ( !warningAlreadyGiven( mappingMethod, selectedMethod ) ) {
```

Comment:
We really should look at redesigning the (error)messages and the way they are handled in mapping-method-trees with nested mappings.  This becomes too tricky to keep track off. Feels to much of localized fix.